CMakeLists.txt

```diff
@@ -77,10 +77,27 @@ add_library(${PROJECT_NAME} INTERFACE)
 #  INTERFACE_COMPILE_FEATURES cxx_std_11
 #)
 
-target_include_directories(${PROJECT_NAME} INTERFACE
-  "$<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include;${PROJECT_SOURCE_DIR}/external/lt>"
-  "$<INSTALL_INTERFACE:include>"
-)
+# Check if there exists a system wise installation of tl-optional
+find_package(tl-optional QUIET)
+
+# if tl-option exists manif will use the system wise installation otherwise it will provide it's own version
+if(tl-optional_FOUND)
+
+  target_include_directories(${PROJECT_NAME} INTERFACE
+    "$<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>"
+    "$<INSTALL_INTERFACE:include>"
+    )
+
+  target_link_libraries(${PROJECT_NAME} INTERFACE tl::optional)
+
+else()
+
+  target_include_directories(${PROJECT_NAME} INTERFACE
+    "$<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include;${PROJECT_SOURCE_DIR}/external/tl>"
+    "$<INSTALL_INTERFACE:include>"
+    )
+
+endif()
```

Comment:
The following is a little more concise. Not tested with an actual system-installed tl-optional,

```suggestion
target_include_directories(${PROJECT_NAME} INTERFACE
  "$<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include;$<$<NOT:$<BOOL:tl-optional_FOUND>>:${PROJECT_SOURCE_DIR}/external/tl>>"
  "$<INSTALL_INTERFACE:include>"
)

# if tl-option exists manif will use the system wise installation otherwise it will provide it's own version
if(tl-optional_FOUND)
  target_link_libraries(${PROJECT_NAME} INTERFACE tl::optional)
endif()
```